compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/TypeReference.kt

```diff
@@ -306,6 +308,42 @@ public sealed class TypeReference {
 
       return (typeElement ?: fail()).requireTypeName()
     }
+
+    private fun findUnwrappedTypesWithTypeAlias(
+      unwrappedTypes: List<TypeReference>
+    ): List<TypeReference> {
+      // For Psi we must inspect the type and check if further arguments have been specified in
+      // a type alias, e.g.
+      //
+      //   typealias MyType<X> = Pair<X, String>
+      //
+      //   val myType: MyType<Int>
+      //
+      // Without inspecting the type, Psi would only tell us that there's on argument `Int`. Type
+      // aliases are pretty much inlined, so we need to treat them specially. The idea is to check
+      // if the type is a type alias and if the type arguments differ. If no, then nothing special
+      // needs to happen. If yes, then we need to merge both type argument lists.
+
+      val children = type.typeElement?.children ?: return unwrappedTypes
+      if (children.size != 2 || children[1] !is KtTypeArgumentList) return unwrappedTypes
```

Comment:
Are we guaranteed this is always exactly 2 for the case we care about? Or should this be `< 2`?

Reply:
There's always two for our case.